game/unicode.cc

```diff
@@ -67,22 +67,24 @@ std::string UnicodeUtil::convertToUTF8 (std::string const& str) {
 	return ss.str();
 }
 
-std::string UnicodeUtil::toLower (std::string const& str, size_t length) {
+std::string UnicodeUtil::toLower (std::string const& str, int length) {
 	std::stringstream ss (str);
 	convertToUTF8 (ss, std::string());
 	std::string ret;
-	if (length == 0 || length >= str.size()) length = str.size();
+	int sizeI = static_cast<int>(str.size());
+	if (length == 0 || length >= sizeI) length = sizeI;
 		icu::UnicodeString tmp = icu::UnicodeString::fromUTF8(ss.str());
 		tmp = tmp.tempSubString(0, length).toLower() + tmp.tempSubString(length, tmp.length() - 1);
 		tmp.toUTF8String(ret);
 	return ret;
 }
 
-std::string UnicodeUtil::toUpper (std::string const& str, size_t length) {
+std::string UnicodeUtil::toUpper (std::string const& str, int length) {
```

Comment:
IMHO, sizes should not be signed

Reply:
I agree butnC libraries do not.

Reply:
well ssize_t exists, but size_t is unsigned by default, not sure I undrstnd your reply

Reply:
Same as above, I'm using int because ICU wants int. I we didn't use int here, we'd have to cast it later anyway.